hack/update-codegen.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# more info: https://cloud.redhat.com/blog/kubernetes-deep-dive-code-generation-customresources
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+cleanup() {
+    echo "cleaning up..."
+    rm -rf ./pkg/github.com/
+    rm -rf api/clustergroupupgradesoperator
+}
+trap "cleanup" EXIT
+
+mkdir -p ./api/clustergroupupgradesoperator
+cd ./api/clustergroupupgradesoperator
+ln -s ../v1alpha1 ./v1alpha1
+cd ../../
+
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd "${SCRIPT_ROOT}"; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
+echo $SCRIPT_ROOT
+echo $CODEGEN_PKG
+
+# TIP: append '-v 10' with the script call below for logs
+bash ${CODEGEN_PKG}/generate-groups.sh "all" \
+    "github.com/openshift-kni/cluster-group-upgrades-operator/pkg/generated" \
+    "github.com/openshift-kni/cluster-group-upgrades-operator/api" \
+    clustergroupupgradesoperator:v1alpha1 \
+    --output-base "./pkg" \
+    --go-header-file hack/boilerplate.go.txt
+
+cp -rp "./pkg/github.com/openshift-kni/cluster-group-upgrades-operator/pkg/" "./pkg/"
```

Comment:
I'm left with a pkg/pkg untracked dir after running this script. This cp command appears to just be copying the generated pkg directory over to the pkg dir.

If I change this to use "./" instead, I think it does what you expect. But now I'm seeing some uncommitted changes in two generated files:
        modified:   pkg/generated/clientset/versioned/fake/register.go
        modified:   pkg/generated/clientset/versioned/scheme/register.go

Adding a call to the update-codegen.sh script to the "generate" target of the Makefile would help catch uncommitted changes in these generated files, as we have CI checks to ensure any such generated changes are committed.

Reply:
Nice! Ok added support for Makefile.

Reply:
Added the CI config. 

`register.go` is coming out differently on macos vs when I run it on rhel. Interesting. The latest pushed code now should be correctly spaced.

Thank you for the `rsync` code :D